js/src/components/tree-select-control/options.js

```diff
@@ -26,6 +26,36 @@ const Options = ( {
 	nodesExpanded = [],
 	onNodesExpandedChange = () => {},
 } ) => {
+	/**
+	 * Verifies if an option is checked.
+	 * An option is checked if their value is selected or all of their children are selected
+	 *
+	 * @param {Option} option The option to verify if is checked
+	 * @return {boolean} True if checked, false otherwise
+	 */
+	const isChecked = ( option ) => {
+		return (
+			value.includes( option.value ) || isEveryChildrenSelected( option )
+		);
+	};
+
+	/**
+	 * Verifies if an option is partially checked.
+	 * An option is partially checked if any of their children is checked or partially checked
+	 *
+	 * @param {Option} parent the Option to verify if its partially checked
+	 * @return {boolean} True if it's partially Checked, false otherwise
+	 */
+	const isPartiallyChecked = ( parent ) => {
+		if ( ! parent.children?.length ) {
+			return false;
+		}
+
+		return parent.children.some(
```

Comment:
Same as the previous comment, my way of thinking was that a Full Checked Option is also partially checked. 

We just prioritize the Full Checked over the Partially Checked. This is mainly to avoid iterating to every node again that could be expensive in there are tons of nodes.

I refactored the naming and added the fully checked status as a param, so it will return as false in case its fully checked. 

See https://github.com/woocommerce/google-listings-and-ads/commit/e7e9081c9335c419846d61b23c9312ea28d5c769

Reply:
Yeah the extra validation helped, another way to do it could be checking if `parent.children.some.length < parent.children.length` &  `parent.children.some.length > 0` and like this you have the same number of iterations and you know exactly it is partially or not.